### ISSUE TITLE
Disable eth_sign by default

### ISSUE THREAD

**Issue by bschorchit** [MetaMask/metamask-mobile#5676]
**Background**

From Harry:
After looking at various phishing kits ([and de-obfuscating](https://twitter.com/sniko_/status/1587882460976713728)), there are enormous trends between different kits that they all abuse `eth_sign` to ask users to inadvertently sign a transaction (as MetaMask UI is opaque to an on-chain transaction being signed). You can read more about the attack [here](https://officercia.medium.com/how-hackers-may-steal-your-ethers-and-why-does-eth-sign-function-matter-69497ee11593).

@409H'd estimate that the abuse of `eth_sign` is near equal to the abuse of Wyvern and Seaport contracts in a manner to steal NFTs through creating orders for 0 eth to a specific beneficiary.

In short: `eth_sign` has been consistently _abused_ by phishers/scammers to steal funds and the case may be that only power-users who use very legacy dapps may need to still use eth_sign.

- Disable `eth_sign` by default - throw an ethErrors.rpc.methodNotFound exception
- Allow the user to enable `eth_sign` through "Advanced Settings"

RPC Method preferences are stored within `rpcMethodPreferences` in the store state, allow for us to develop this further if someone wants to lock down rpc calls into MetaMask.

**Comment by Cal-L** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/MetaMask-Mobile-5f984938ddc0e4001d4b79cb/issues/MetaMask/metamask-mobile/5676?planning-poker) with Zenhub @Fatxx @tommasini @jpcloureiro

**Comment by Cal-L** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/MetaMask-Mobile-5f984938ddc0e4001d4b79cb/issues/MetaMask/metamask-mobile/5676?planning-poker) with Zenhub @chrisleewilcox

**Comment by bschorchit** [comment]
Great question, Cal. It fails silently to the user and this is an intended behavior. 
We'll revaluate wether we want to include a recovery path to the user in the future as we gather user and dapp feedback from this change. But right now it's intentional to not provide a recovery path as this method is mainly being used by scammers.

**Comment by Cal-L** [comment]
cc @yanrong-chen for design support

**Comment by Cal-L** [comment]
@holantonela 
 When you have the chance - could you update the ticket to reference Figma with the mobile UI? :thank-you: I think we're clear from the Product aspect but it would be great @bschorchit if you could take a look in case we missed anything. Thanks!

**Comment by bschorchit** [comment]
I made a slight update to the above to note that this should be the last toggle on Advanced Settings to match parity with Extension. Below is a screenshot of extension prod for reference as well.
<img width="370" alt="Screenshot 2023-02-15 at 13 53 22" src="https://user-images.githubusercontent.com/34968356/219098755-e46ae13c-e58f-4b13-841f-34efa2cab2ca.png">

**Comment by yanrong-chen** [comment]
Design for mobile https://www.figma.com/file/X5tDiDfJN8IA9TXA1gM95S/Add-option-to-opt-out-(Disable-eth_sign-by-default)?node-id=0%3A1&t=9u0CJbbBiGfPB0JE-0

**Comment by vandan** [comment]
On request for DevEx involvement here..
@bschorchit Given the benefits of deprecating eth_sign and urgency, I would like to confirm if we're scoping this to the minimum viable solution. For example, is it essential for Mobile to have precise parity with Extension on the "Fallback" option where we:
* Allow the user to enable eth_sign through "Advanced Settings" (which requires UI changes)

Would it be that problematic to only offer the enabling of eth_sign on Extension?

**Comment by vandan** [comment]
Confirmed with @rekmarks that we should aim for parity with Extension because the effort to implement the mobile UI reasonable for DevEx (still react-based with pre-existing design components).
